src/mkdir/mkdir.rs

```diff
@@ -93,13 +93,7 @@ fn exec(dirs: Vec<String>, recursive: bool, mode: u16, verbose: bool) -> i32 {
     for dir in &dirs {
         let path = Path::new(dir);
         if recursive {
-            let mut pathbuf = PathBuf::new();
-            for component in path.components() {
-                pathbuf.push(component.as_os_str());
-                if !path.is_dir() {
-                    status |= mkdir(pathbuf.as_path(), mode, verbose);
-                }
-            }
+            status |= mkdir(path, recursive, mode, verbose);
```

Comment:
Since `status |= mkdir(path, recursive, mode, verbose);` appears so much:
```rust
for dir in &dirs {
    let path = Path::new(dir);
    if !recursive {
        if let Some(parent) = path.parent() {
            if parent != empty && !parent.exists() {
                show_info!(
                    "cannot create directory '{}': No such file or directory",
                    path.display()
                );
                status = 1;
                continue;
            }
        }
    }
    status |= mkdir(path, recursive, mode, verbose);
}
```